File: src/main/java/de/hannisoft/gaveplan/model/Point.java
package de.hannisoft.graveplan.model;

public class Point {
    private final int id;
    private int x;
    private int y;

    public Point(int id, int x, int y) {
        this.id = id;
        this.x = x;
        this.y = y;
    }

    public int getId() {
        return id;
    }

    public int getX() {
        return x;
    }

    public int getY() {
        return y;
    }

    public void rotate(double degree) {
        if (degree == 0) {
            return;
        }
        double xOld = x;
        double yOld = y;
        double xd = xOld * Math.cos(degree) - yOld * Math.sin(degree);
        x = Long.valueOf(Math.round(xd)).intValue();
        double yd = xOld * Math.sin(degree) + yOld * Math.cos(degree);
        y = Long.valueOf(Math.round(yd)).intValue();
    }

    public void scale(double xFactor, double yFactor) {
        x = Long.valueOf(Math.round(x * xFactor)).intValue();
        y = Long.valueOf(Math.round(y * yFactor)).intValue();
    }

    public void move(int xDelta, int yDelta) {
        x = x + xDelta;
        y = y + yDelta;
    }

    @Override
    public String toString() {
        return id + "=" + x + "," + y;
    }
}

File: src/main/kotlin/de/hannisoft/graveplan/model/Point.kt
package de.hannisoft.graveplan.model

import kotlin.math.cos
import kotlin.math.roundToInt
import kotlin.math.sin

data class Point(
    val id: Int,
    var x: Int,
    var y: Int
) {

    fun rotate(degree: Double) {
        if (degree == 0.0) return

        val xOld = x.toDouble()
        val yOld = y.toDouble()

        val xd = xOld * cos(degree) - yOld * sin(degree)
        val yd = xOld * sin(degree) + yOld * cos(degree)

        x = xd.roundToInt()
        y = yd.roundToInt()
    }

    fun scale(xFactor: Double, yFactor: Double) {
        x = (x * xFactor).roundToInt()
        y = (y * yFactor).roundToInt()
    }

    fun move(xDelta: Int, yDelta: Int) {
        x += xDelta
        y += yDelta
    }

    override fun toString(): String = "$id=$x,$y"
}